Storage/SQL/SQLiteHistory.swift

```diff
@@ -277,37 +285,49 @@ fileprivate struct SQLiteFrecentHistory : FrecentHistory {
         }
 
         // Innermost: grab history items and basic visit/domain metadata.
-        var ungroupedSQL =
-            "SELECT \(TableHistory).id AS historyID, \(TableHistory).url AS url, \(TableHistory).title AS title, \(TableHistory).guid AS guid, domain_id, domain" +
-                ", COALESCE(max(case \(TableVisits).is_local when 1 then \(TableVisits).date else 0 end), 0) AS localVisitDate" +
-                ", COALESCE(max(case \(TableVisits).is_local when 0 then \(TableVisits).date else 0 end), 0) AS remoteVisitDate" +
-                ", COALESCE(sum(\(TableVisits).is_local), 0) AS localVisitCount" +
-                ", COALESCE(sum(case \(TableVisits).is_local when 1 then 0 else 1 end), 0) AS remoteVisitCount" +
-                " FROM \(TableHistory) " +
-                "INNER JOIN \(TableDomains) ON \(TableDomains).id = \(TableHistory).domain_id " +
-        "INNER JOIN \(TableVisits) ON \(TableVisits).siteID = \(TableHistory).id "
+        var ungroupedSQL = """
+            SELECT \(TableHistory).id AS historyID, \(TableHistory).url AS url,
+                \(TableHistory).title AS title, \(TableHistory).guid AS guid, domain_id, domain,
+                COALESCE(MAX(CASE \(TableVisits).is_local WHEN 1 THEN \(TableVisits).date ELSE 0 END), 0) AS localVisitDate,
+                COALESCE(MAX(CASE \(TableVisits).is_local WHEN 0 THEN \(TableVisits).date ELSE 0 END), 0) AS remoteVisitDate,
+                COALESCE(SUM(\(TableVisits).is_local), 0) AS localVisitCount,
+                COALESCE(SUM(CASE \(TableVisits).is_local WHEN 1 THEN 0 ELSE 1 END), 0) AS remoteVisitCount
+            FROM \(TableHistory)
+                INNER JOIN \(TableDomains) ON
+                    \(TableDomains).id = \(TableHistory).domain_id
+                INNER JOIN \(TableVisits) ON
+                    \(TableVisits).siteID = \(TableHistory).id
+            """
 
         if includeBookmarks {
-            ungroupedSQL.append("LEFT JOIN \(ViewAllBookmarks) on \(ViewAllBookmarks).url = \(TableHistory).url ")
+            ungroupedSQL.append(" LEFT JOIN \(ViewAllBookmarks) ON \(ViewAllBookmarks).url = \(TableHistory).url")
```

Comment:
Ooh, is that a footgun in multi-line literals — automatically trimmed?

Reply:
Yes. See also why I put SQL comments on the previous line.